src/urllib3/request.py

```diff
@@ -88,12 +88,14 @@ def request(
 
         if json is not None and body is not None:
             raise TypeError(
-                "request got values for both 'json' and 'body', can only specify one. Prefer json"
+                "request got values for both 'json' and 'body', can only specify one."
             )
 
         if json is not None:
             if headers is None:
-                headers = self.headers.setdefault("Content-Type", "application/json")  # type: ignore
+                headers = self.headers
+                headers["Content-Type"] = "application/json"
```

Comment:
We probably also want to provide a Content-Type header even if headers are passed in _but_ Content-Type is not passed in. Ideally in a case-insensitive way. :)

Maybe normalizing headers to a HTTPHeaderDict early on is a good strategy which would take care of normalizing the keys? Not sure!

(A test for this would be great)